app/src/main/java/com/example/topmovies/fragment/MovieDetailsFragment.kt

```diff
@@ -4,29 +4,49 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import com.bumptech.glide.Glide
 import com.example.topmovies.databinding.FragmentDetailsMovieBinding
-import com.example.topmovies.viewmodel.MovieViewModel
-import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+import com.example.topmovies.viewmodel.MovieDetailsViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+const val FRAGMENT_KEY = "movieID"
 
 class MovieDetailsFragment : BaseFragment() {
     
-    private val binding by lazy { FragmentDetailsMovieBinding.inflate(layoutInflater) }
-    private val movieViewModel by sharedViewModel<MovieViewModel>()
-    
-    companion object {
-        
-        const val FRAGMENT_KEY = "movieID"
-    }
+    private var _binding: FragmentDetailsMovieBinding? = null
+    private val binding get() = _binding!!
+    private val movieViewModel: MovieDetailsViewModel by viewModel()
     
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
-    ) = binding.root
+    ): View {
+        _binding = FragmentDetailsMovieBinding.inflate(inflater, container, false)
+        return binding.root
+    }
     
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        showBackButton()
+        arguments?.getString(FRAGMENT_KEY)?.let { loadMovieDetailsById(it) }
+        setupViewModel()
         setupUI()
-        requireArguments().getString(FRAGMENT_KEY)?.let { loadMovieDetailsById(it) }
+    }
+    
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+    
+    private fun setupViewModel() {
+        movieViewModel.movieDetails.observe(viewLifecycleOwner) {
+            requireArguments().getString(FRAGMENT_KEY)?.let { loadMovieDetailsById(it) }
```

Comment:
Why you are doing this one?
As it seems to me - you'll end up in an endless loop of the same request.